eth/beacon/types/shard_and_committees.py

```diff
@@ -18,19 +18,19 @@ class ShardAndCommittee(rlp.Serializable):
     Note: using RLP until we have standardized serialization format.
     """
     fields = [
-        # The shard ID
-        ('shard_id', int16),
+        # Shard number
+        ('shard', uint64),
         # Validator indices
-        ('committee', CountableList(int24)),
+        ('committee', CountableList(uint24)),
     ]
 
     def __init__(self,
-                 shard_id: int,
+                 shard: int,
                  committee: Iterable[int])-> None:
```

Comment:
I think `Sequence` is our default generic iterable unless we specifically want to support generators (which we do sometimes but only in cases where we're sure the code is designed to handle the value as if it was a generator).  I think this will actually work correctly if you pass in a generator, but I don't think the `rlp.Serializable` API is tested to support that.